tests/data_classes.py

```diff
@@ -0,0 +1,642 @@
+        # 2 column - NAME, IP4
+        # In case of releases, it will only be one # NAME
+        data = self.standard_parse()
+        if len(data) == 1:
+            self.release = data[0]
+        else:
+            self.name, self.ip4 = data
+
+    def df_parse(self):
+        self.name, self.crt, self.res, self.qta, self.use, self.ava = \
+            self.standard_parse()
+
+    def snapshot_parse(self):
+        self.name, self.created, self.rsize, self.used = self.standard_parse()
+
+
+class ZFS:
+    # TODO: Improve how we manage zfs object here
+    pool = None
+    pool_mountpoint = None
+
+    def __init__(self):
+        if not self.pool:
+            with libzfs.ZFS() as zfs:
+                pools = [
+                    p for p in zfs.pools
+                    if p.root_dataset.__getstate__().get(
+                        'properties'
+                    ).get(
+                        'org.freebsd.ioc:active', {}
+                    ).get('value', 'no') == 'yes'
+                ]
+
+                assert len(pools) == 1, f'{len(pools)} Active pools found'
+
+                ZFS.pool = pools[0].name
+                ZFS.pool_mountpoint = pools[0].root_dataset.mountpoint
+
+    @staticmethod
+    def get_dataset(identifier):
+        with libzfs.ZFS() as zfs:
+            return zfs.get_dataset(identifier).__getstate__()
+
+    @staticmethod
+    def get(identifier):
+        with libzfs.ZFS() as zfs:
+            return zfs.get(identifier).__getstate__()
+
+    def _zfs_get_properties(self, identifier):
+        if '/' in identifier:
+            dataset = self.get_dataset(identifier)
+
+            return dataset['properties']
+        else:
+            pool = self.get(identifier)
+
+            return pool['root_dataset']['properties']
+
+    def zfs_get_property(self, identifier, key):
+        try:
+            return self._zfs_get_properties(identifier)[key]['value']
+        except Exception:
+            return '-'
+
+    def get_dataset_safely(self, dataset_name):
+        try:
+            return self.get_dataset(dataset_name)
+        except libzfs.ZFSException:
+            pass
+
+    @staticmethod
+    def get_snapshots_recursively(dataset_name):
+        with libzfs.ZFS() as zfs:
+            return zfs.get_dataset(dataset_name).__getstate__(
+                snapshots_recursive=True
+            )['snapshots_recursive']
+
+    @staticmethod
+    def get_snapshots(dataset_name):
+        with libzfs.ZFS() as zfs:
+            return zfs.get_dataset(dataset_name).__getstate__(
+                snapshots=True
+            )['snapshots']
+
+    @staticmethod
+    def get_snapshot_safely(snap):
+        try:
+            with libzfs.ZFS() as zfs:
+                return zfs.get_snapshot(snap).__getstate__()
+        except libzfs.ZFSException:
+            pass
+
+    @property
+    def iocage_dataset(self):
+        return self.get_dataset(f'{self.pool}/iocage')
+
+    @property
+    def releases_dataset(self):
```

Comment:
Since we have all of these, we should add a `jails_dataset` method as well

Reply:
We cover that in the jails class ;)